client/src/components/card.tsx

```diff
@@ -1,8 +1,21 @@
 import { useState, useEffect, useRef } from 'react';
 import { Link, useNavigate } from 'react-router-dom';
 import axios from 'axios';
+import Modal from './modal'
+import { IAppointment } from '../../../server/interfaces/appointment.interface';
+interface IProps extends IAppointment {
+  _id?: string;
+  setShowModal?: React.Dispatch<React.SetStateAction<boolean>>
+  setValue?: React.Dispatch<React.SetStateAction<boolean>>
+}
 
-const Card: React.FC = () => {
+const Card: React.FC<IProps> = (props) => {
+  const { setShowModal, setValue, _id, name, specialty, time, location } = props;
+  const [modalDetails, setModalDetails] = useState({})
```

Comment:
can nix here through line 18 since we're taking a different direction for modal. can also take out `setShowModal` in props destructure on line 13